theano/gpuarray/tests/test_dnn.py

```diff
@@ -1611,6 +1611,39 @@ def test_dnn_reduction_absmax():
         utt.assert_allclose(np.max(np.abs(M_val), axis=axis), f(M_val))
 
 
+def test_dnn_reduction_axis_size_one():
+    if not dnn.dnn_available(test_ctx_name) or dnn.version(raises=False) < 6000:
+        raise SkipTest(dnn.dnn_available.msg)
+    for dtype in ('float16', 'float32', 'float64'):
+        for shape, axis in [[(1, 2, 3), 0], [(2, 1, 3), 1], [(2, 3, 1), 2]]:
```

Comment:
Can you test with multiple reduction axis? like shape (1,2,3,4) and axis 0,2 and shape (1,2,3,1) and axis 0 and 3.

Reply:
Updated.

**NB**: When trying to test multiple axes, I got some optimizations issues. Sometimes acc_dtype is None again. Sometimes, when reduced axes are consecutive, then there is an intermediate optimization for GpuCaReduceCuda, so that GpuDnnReduction is not introduced as expected. And sometimes the reduction pattern (e.g. `1010`) is not supported by GpuCaReduceCuda. I finally choosed patterns that are supported (`101`, `0101`).

Reply:
So it mean we don't always introduce it. The good fix would be to change how this is optimized, but we won't do it. The only problem is a bigger graph then needed.